azurerm/resource_arm_maps_account.go

```diff
@@ -0,0 +1,186 @@
+				Type:             schema.TypeString,
+				Optional:         true,
+				ForceNew:         true,
+				Default:          "S0",
+				DiffSuppressFunc: suppress.CaseDifference,
+				ValidateFunc: validation.StringInSlice([]string{
+					"S0",
+					"S1",
+				}, true),
+			},
+
+			"tags": tagsSchema(),
+
+			"x_ms_client_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"primary_access_key": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+
+			"secondary_access_key": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+		},
+	}
+}
+
+func resourceArmMapsAccountCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).maps.AccountsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for AzureRM Maps Account creation.")
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	location := globalLocation()
+	tags := d.Get("tags").(map[string]interface{})
+	sku := d.Get("sku").(string)
+
+	if requireResourcesToBeImported && d.IsNewResource() {
+		existing, err := client.Get(ctx, resGroup, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing Maps Account %q (Resource Group %q): %+v", name, resGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_maps_account", *existing.ID)
+		}
+	}
+
+	parameters := maps.AccountCreateParameters{
+		Location: &location,
+		Sku: &maps.Sku{
+			Name: &sku,
+		},
+		Tags: expandTags(tags),
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resGroup, name, parameters); err != nil {
+		return fmt.Errorf("Error creating/updating Maps Account %q (Resource Group %q) %+v", name, resGroup, err)
+	}
+
+	read, err := client.Get(ctx, resGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Maps Account %q (Resource Group %q) %+v", name, resGroup, err)
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read Maps Account %q (Resource Group %q) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmMapsAccountRead(d, meta)
+}
+
+func resourceArmMapsAccountRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).maps.AccountsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["accounts"]
+
+	resp, err := client.Get(ctx, resGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error making Read request on Maps Account %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resGroup)
+	if sku := resp.Sku; sku != nil {
+		d.Set("sku", string(*sku.Name))
```

Comment:
d.Set deals with nil values so could we do this instead:
```suggestion
		d.Set("sku_name", sku.Name)
```